core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -180,4 +180,21 @@ public byte[] toArray() {
   public String toString() {
     return new String(data, offset, length, UTF_8);
   }
+
+  public void setArray(byte[] data, int offset, int length) {
```

Comment:
Feel free to ignore this comment as its personal preference.  I like the method name `reset` used in #4735, so could use that instead of `setArray`. This is a new public API method is being introduced, so can call it whatever we like.  Do not need to use the same method name as MutableByteSequence

Reply:
Sounds good to me, I renamed `setArray` to `reset`.